client/internal/netflow/manager.go

```diff
@@ -124,7 +124,9 @@ func (m *Manager) disableFlow() error {
 	m.logger.Close()
 
 	if m.receiverClient != nil {
-		return m.receiverClient.Close()
+		clientErr := m.receiverClient.Close()
+		m.receiverClient = nil
+		return clientErr
```

Comment:
Hold on, we need to check if this is nil, otherwise we might return a non-nil interface containing nil

```suggestion
		client := m.receiverClient
		m.receiverClient = nil
		if err := client.Close(); err != nil {
				return err
		}
		return nil
```